src/fs/fs.go

```diff
@@ -119,3 +120,31 @@ func IsDirectory(path string) bool {
 	info, err := os.Stat(path)
 	return err == nil && info.IsDir()
 }
+
+// Memoize this to cut down on filesystem operations
+var isPackageMemo = map[string]bool{}
+var isPackageMutex sync.RWMutex
+
+// IsPackage returns true if the given directory name is a package (i.e. contains a build file)
+func IsPackage(buildFileNames []string, name string) bool {
+	isPackageMutex.RLock()
```

Comment:
Noted that we had this before, thanks for preserving.... but on further thought do we think it's worth it? Do we really re-glob over enough package directories to make it worth it?